app/api/sponsors.py

```diff
@@ -57,35 +70,18 @@ def query(self, view_kwargs):
         query_ = self.session.query(Sponsor)
         if view_kwargs.get('event_id'):
             event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            query_ = query_.join(Event).filter(Event.id == event.id)
+            query_ = query_.join(Event).filter(Event.id == event.id, Event.state == 'published')
```

Comment:
@shubham-padia Event Organiser can see all the sponsors and not only for `published`. Please check again!